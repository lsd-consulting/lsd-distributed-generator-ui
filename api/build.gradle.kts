plugins {
    kotlin("jvm")
    `maven-publish`
    id("java-library")
    id("signing")
    id("kotlin-kapt")
}

publishing {
    publications {
        create<MavenPublication>("mavenJava") {
            groupId = "$group"
            artifactId = "${rootProject.name}-${project.name}"
            version = rootProject.version.toString()

            from(components["java"])
            pom {
                name.set("lsd-distributed-generator-ui")
                description.set("This is the graphical interface providing the LSD generation functionality.")
                url.set("https://github.com/lsd-consulting/lsd-distributed-generator-ui")
                licenses {
                    license {
                        name.set("MIT License")
                        url.set("https://github.com/lsd-consulting/lsd-distributed-generator-ui/blob/main/LICENSE.txt")
                        distribution.set("repo")
                    }
                }
                developers {
                    developer {
                        name.set("Lukasz")
                        email.set("lukasz.gryzbon@gmail.com")
                        organization.set("Integreety Ltd.")
                        organizationUrl.set("https://www.integreety.co.uk")
                    }
                    developer {
                        name.set("Nick")
                        email.set("nicholas.mcdowall@gmail.com")
                        organization.set("NKM IT Solutions")
                        organizationUrl.set("https://github.com/nickmcdowall")
                    }
                }
                scm {
                    url.set("https://github.com/lsd-consulting/lsd-distributed-generator-ui.git")
                }
            }
            repositories {
                maven {
                    name = "sonatype"
                    url = uri("https://ossrh-staging-api.central.sonatype.com/service/local/staging/deploy/maven2/")
                    credentials(PasswordCredentials::class)
                }
            }
        }
    }
}

signing {
    project.findProperty("signingKey")?.let {
        // Use in-memory ascii-armored keys
        val signingKey: String? by project
        val signingPassword: String? by project
        useInMemoryPgpKeys(signingKey, signingPassword)
        sign(publishing.publications["mavenJava"])
    } ?: run {
        sign(publishing.publications["mavenJava"])
    }
}
