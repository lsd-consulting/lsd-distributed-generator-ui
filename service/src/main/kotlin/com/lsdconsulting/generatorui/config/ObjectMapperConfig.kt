package com.lsdconsulting.generatorui.config

import com.fasterxml.jackson.databind.ObjectMapper
import com.lsdconsulting.generatorui.config.objectmapper.ObjectMapperCreator
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter

@Configuration
class ObjectMapperConfig {

    @Bean
    fun objectMapper(): ObjectMapper = ObjectMapperCreator().create()

    @Bean
    fun converter(objectMapper: ObjectMapper): MappingJackson2HttpMessageConverter? {
        val converter = MappingJackson2HttpMessageConverter()
        converter.objectMapper = objectMapper
        return converter
    }
}
