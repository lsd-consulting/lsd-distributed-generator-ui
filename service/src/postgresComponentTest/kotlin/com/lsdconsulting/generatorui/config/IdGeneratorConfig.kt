package com.lsdconsulting.generatorui.config

import com.lsd.core.IdGenerator
import org.springframework.boot.test.context.TestConfiguration
import org.springframework.context.annotation.Bean

@TestConfiguration
class IdGeneratorConfig {
    @Bean
    fun idGenerator() = IdGenerator(true)
}
