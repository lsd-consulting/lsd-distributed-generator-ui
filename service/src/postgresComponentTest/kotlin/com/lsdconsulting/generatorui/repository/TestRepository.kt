package com.lsdconsulting.generatorui.repository

import org.springframework.stereotype.Component
import javax.sql.DataSource

@Component
class TestRepository {
    fun createTable(dataSource: DataSource) {
        val prepareDatabaseQuery = javaClass.getResourceAsStream("/db/prepareDatabase.sql")?.bufferedReader()?.readText()
        dataSource.connection.use { con ->
            con.prepareStatement(prepareDatabaseQuery).use { pst ->
                pst.executeUpdate()
            }
        }
    }

    fun clearTable(dataSource: DataSource) {
        dataSource.connection.use { con ->
            con.prepareStatement("truncate lsd.intercepted_interactions").use { pst ->
                pst.executeUpdate()
            }
        }
    }
}
