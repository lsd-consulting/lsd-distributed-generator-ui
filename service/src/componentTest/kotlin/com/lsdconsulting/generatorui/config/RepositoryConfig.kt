package com.lsdconsulting.generatorui.config

import com.lsdconsulting.generatorui.repository.TestRepository
import com.lsdconsulting.generatorui.repository.TestRepository.Companion.setupDatabase
import org.springframework.boot.test.context.TestConfiguration
import org.springframework.context.annotation.Bean

@TestConfiguration
class RepositoryConfig {
    companion object {
        // This is because the configs in spring.factories run always before any test configs.
        init {
            setupDatabase()
        }
    }

    @Bean
    fun testRepository(): TestRepository {
        return TestRepository()
    }
}
