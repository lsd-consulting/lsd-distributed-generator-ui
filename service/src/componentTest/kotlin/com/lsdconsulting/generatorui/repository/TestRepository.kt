package com.lsdconsulting.generatorui.repository

import com.lsdconsulting.generatorui.config.logger.log
import com.mongodb.ConnectionString
import com.mongodb.MongoClientSettings
import com.mongodb.client.MongoClient
import com.mongodb.client.MongoClients
import de.flapdoodle.embed.mongo.MongodExecutable
import de.flapdoodle.embed.mongo.MongodStarter
import de.flapdoodle.embed.mongo.config.MongodConfig
import de.flapdoodle.embed.mongo.config.Net
import de.flapdoodle.embed.mongo.distribution.Version.Main.PRODUCTION
import de.flapdoodle.embed.process.runtime.Network.localhostIsIPv6
import java.io.IOException

class TestRepository {
//    private val pojoCodecRegistry: CodecRegistry = CodecRegistries.fromRegistries(
//        MongoClientSettings.getDefaultCodecRegistry(),
//        CodecRegistries.fromCodecs(ZonedDateTimeCodec(), TypeCodec()),
//        CodecRegistries.fromProviders(PojoCodecProvider.builder().automatic(true).build())
//    )
//    val collection: MongoCollection<Document>
//        get() {
//            val database: MongoDatabase = mongoClient.getDatabase(DATABASE_NAME)
//            return database.getCollection(COLLECTION_NAME).withCodecRegistry(pojoCodecRegistry)
//        }

//    fun findAll(traceId: String?): List<InterceptedInteraction> {
//        log().info("Retrieving interceptedInteractions for traceId:{}", traceId)
//        val database: MongoDatabase = mongoClient.getDatabase(DATABASE_NAME)
//        val collection: MongoCollection<Document> =
//            database.getCollection(COLLECTION_NAME).withCodecRegistry(pojoCodecRegistry)
//        val result: MutableList<InterceptedInteraction> = ArrayList<InterceptedInteraction>()
//        try {
//            collection.find(eq("traceId", traceId), InterceptedInteraction::class.java)
//                .iterator().use { cursor ->
//                while (cursor.hasNext()) {
//                    val interceptedInteraction: InterceptedInteraction = cursor.next()
//                    log().info("Retrieved interceptedInteraction:{}", interceptedInteraction)
//                    result.add(interceptedInteraction)
//                }
//            }
//        } catch (e: MongoException) {
//            log().error(
//                "Failed to retrieve interceptedInteractions - message:{}, stackTrace:{}",
//                e.message,
//                e.stackTrace
//            )
//        }
//        return result
//    }

    companion object {
        private const val MONGODB_HOST = "localhost"
        private const val MONGODB_PORT = 27017
        private const val DATABASE_NAME = "lsd"
        private const val COLLECTION_NAME = "interceptedInteraction"
        private lateinit var mongoClient: MongoClient
        private lateinit var mongodExecutable: MongodExecutable

        fun setupDatabase() {
            try {
                val mongodConfig: MongodConfig = MongodConfig.builder()
                    .version(PRODUCTION)
                    .net(Net(MONGODB_HOST, MONGODB_PORT, localhostIsIPv6()))
                    .build()
                mongodExecutable = MongodStarter.getDefaultInstance().prepare(mongodConfig)
                mongodExecutable.start()
                mongoClient = MongoClients.create(
                    MongoClientSettings.builder()
                        .applyConnectionString(ConnectionString("mongodb://$MONGODB_HOST:$MONGODB_PORT"))
                        .retryWrites(true)
                        .build()
                )
            } catch (e: IOException) {
                log().error(e.message, e)
            }
        }

        fun tearDownDatabase() {
            mongoClient.close()
            mongodExecutable.stop()
        }
    }
}
