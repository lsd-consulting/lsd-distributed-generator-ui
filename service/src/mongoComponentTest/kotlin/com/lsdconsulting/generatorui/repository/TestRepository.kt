package com.lsdconsulting.generatorui.repository

import com.lsdconsulting.generatorui.config.logger.log
import com.mongodb.ConnectionString
import com.mongodb.MongoClientSettings
import com.mongodb.client.MongoClient
import com.mongodb.client.MongoClients
import de.flapdoodle.embed.mongo.MongodExecutable
import de.flapdoodle.embed.mongo.MongodStarter
import de.flapdoodle.embed.mongo.config.MongodConfig
import de.flapdoodle.embed.mongo.config.Net
import de.flapdoodle.embed.mongo.distribution.Version.Main.V5_0
import de.flapdoodle.embed.process.runtime.Network.localhostIsIPv6
import java.io.IOException

class TestRepository {
    companion object {
        private const val MONGODB_HOST = "localhost"
        private const val MONGODB_PORT = 27017
        private lateinit var mongoClient: MongoClient
        private lateinit var mongodExecutable: MongodExecutable

        fun setupDatabase() {
            try {
                val mongodConfig: MongodConfig = MongodConfig.builder()
                    .version(V5_0)
                    .net(Net(MONGODB_HOST, MONGODB_PORT, localhostIsIPv6()))
                    .build()
                mongodExecutable = MongodStarter.getDefaultInstance().prepare(mongodConfig)
                mongodExecutable.start()
                mongoClient = MongoClients.create(
                    MongoClientSettings.builder()
                        .applyConnectionString(ConnectionString("mongodb://$MONGODB_HOST:$MONGODB_PORT"))
                        .retryWrites(true)
                        .build()
                )
            } catch (e: IOException) {
                log().error(e.message, e)
            }
        }

        fun tearDownDatabase() {
            mongoClient.close()
            mongodExecutable.stop()
        }
    }
}
