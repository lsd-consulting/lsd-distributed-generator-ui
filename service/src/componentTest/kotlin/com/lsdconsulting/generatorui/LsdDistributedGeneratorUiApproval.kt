package com.lsdconsulting.generatorui

import com.lsdconsulting.generatorui.config.RepositoryConfig
import com.lsdconsulting.generatorui.repository.TestRepository
import io.lsdconsulting.lsd.distributed.access.model.InteractionType
import io.lsdconsulting.lsd.distributed.access.model.InteractionType.REQUEST
import io.lsdconsulting.lsd.distributed.access.model.InteractionType.RESPONSE
import io.lsdconsulting.lsd.distributed.access.model.InterceptedInteraction
import io.lsdconsulting.lsd.distributed.access.repository.InterceptedDocumentRepository
import io.lsdconsulting.lsd.distributed.mongo.config.LibraryConfig
import org.approvaltests.Approvals
import org.junit.jupiter.api.AfterAll
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.context.SpringBootTest.WebEnvironment.DEFINED_PORT
import org.springframework.boot.test.web.client.TestRestTemplate
import org.springframework.boot.test.web.client.getForEntity
import org.springframework.context.annotation.Import
import org.springframework.http.ResponseEntity
import org.springframework.test.context.TestPropertySource
import java.time.Instant.EPOCH
import java.time.ZoneId
import java.time.ZonedDateTime

@SpringBootTest(webEnvironment = DEFINED_PORT, classes = [LsdGeneratorUiApplication::class])
@TestPropertySource("classpath:application-test.properties")
@Import(RepositoryConfig::class, LibraryConfig::class)
class LsdGeneratorUiApproval(
    @Autowired private val interceptedDocumentRepository: InterceptedDocumentRepository,
    @Autowired private val testRestTemplate: TestRestTemplate
) {

    val traceId = "someTraceId"
    val body = "someBody"
    val requestHeaders = mapOf("header" to listOf("value"))
    val responseHeaders = mapOf("header" to listOf("value"))
    val httpMethod = "GET"
    val profile = "TEST"
    val elapsedTime = 25L
    val createdAt: ZonedDateTime = ZonedDateTime.ofInstant(EPOCH, ZoneId.of("UTC"))

    companion object {
        @BeforeAll
        @JvmStatic
        internal fun beforeAll() {
            TestRepository.setupDatabase()
        }

        @AfterAll
        @JvmStatic
        internal fun afterAll() {
            TestRepository.tearDownDatabase()
        }
    }

    @Test
    fun shouldReturnApprovedResponse() {
        // Remove the LSD interceptors
        testRestTemplate.restTemplate.interceptors.clear()

        interceptedDocumentRepository.save(buildInterceptedInteraction(null, REQUEST))
        interceptedDocumentRepository.save(buildInterceptedInteraction("200", RESPONSE))

        val result: ResponseEntity<String> = testRestTemplate.getForEntity("/lsd/$traceId")

        Approvals.verifyHtml(result.body)
    }

    private fun buildInterceptedInteraction(
        status: String?,
        type: InteractionType
    ): InterceptedInteraction? = InterceptedInteraction.builder()
        .traceId(traceId)
        .body(body)
        .requestHeaders(requestHeaders)
        .responseHeaders(responseHeaders)
        .serviceName("Source")
        .target("Target")
        .path("/path")
        .httpStatus(status)
        .httpMethod(httpMethod)
        .interactionType(type)
        .profile(profile)
        .elapsedTime(elapsedTime)
        .createdAt(createdAt)
        .build()
}