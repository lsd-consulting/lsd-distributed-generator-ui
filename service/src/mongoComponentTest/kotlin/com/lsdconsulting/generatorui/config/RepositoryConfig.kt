package com.lsdconsulting.generatorui.config

import com.lsdconsulting.generatorui.repository.TestRepository
import org.springframework.boot.test.context.TestConfiguration
import org.springframework.context.annotation.Bean

@TestConfiguration
class RepositoryConfig {
    @Bean
    fun testRepository(): TestRepository {
        return TestRepository()
    }
}
