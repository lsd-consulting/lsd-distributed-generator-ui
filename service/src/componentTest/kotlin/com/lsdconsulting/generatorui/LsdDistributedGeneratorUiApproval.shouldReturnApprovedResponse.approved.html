<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xmlns="http://www.w3.org/1999/html" xml:lang="en" lang="en">
<head>
    <title>Diagram for traceIds: someTraceId</title>
    <link rel="stylesheet" href="https://lsd-consulting.github.io/lsd-core/src/main/resources/static/style.css">
</head>
<body>
<main role="main">
    <h1 class="logo">LSD</h1>
    <h1>Diagram for traceIds: someTraceId</h1>
    <p>Timeframe: 1970-01-01T00:00Z[UTC] - 1970-01-01T00:00Z[UTC]</p>
    <article id="3">
                <aside class="facts">
            <div class="factTable">
                <h3>Key Facts</h3>
                <div class="factRow">
                    <div class="factRowHead">Description</div>
                    <div class="factRowHead">Value</div>
                </div>
                                <div class="factRow">
                    <div class="factCell">traceIds</div>
                    <div class="factCell">someTraceId</div>
                </div>
                            </div>
        </aside>
        <section class="sequence diagram svg ">
            <h3 class="">Sequence Diagram</h3>
            <pre><?xml version="1.0" encoding="us-ascii" standalone="no"?><svg xmlns="http://www.w3.org/2000/svg"
                                                                               contentStyleType="text/css" height="550px" preserveAspectRatio="none" style="width:304px;height:550px;background:#000000;" version="1.1" viewBox="0 0 304 550" width="304px" zoomAndPan="magnify"><defs/><g><rect fill="#0D2413" height="1" style="stroke:#0D2413;stroke-width:1.0;" width="1" x="0" y="0"/><text fill="#33FF02" font-family="sans-serif" font-size="12" font-style="italic" font-weight="bold" lengthAdjust="spacing" textLength="116" x="5" y="17">PlantUML 1.2023.5</text><rect fill="#33FF02" height="21.4883" style="stroke:#33FF02;stroke-width:1.0;" width="292" x="5" y="27.1328"/><text fill="#000000" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="165" x="6" y="42.1328">[From string (line 12) ]</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="62.6211">&#160;</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="74" x="5" y="79.1094">@startuml</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="95.5977">&#160;</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="112.0859">&#160;</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="128.5742">&#160;</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="145.0625">&#160;</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="9" x="5" y="161.5508">...</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="169" x="5" y="178.0391">... ( skipping 132 lines )</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="9" x="5" y="194.5273">...</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="168" x="15" y="211.0156">BackgroundColor white</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="129" x="15" y="227.5039">BorderColor black</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="243.9922">}</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="178" x="5" y="260.4805">skinparam StereotypeN {</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="168" x="15" y="276.9688">BackgroundColor white</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="129" x="15" y="293.457">BorderColor black</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="309.9453">}</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="231" x="5" y="326.4336">skinparam UseCaseStereoType {</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="114" x="15" y="342.9219">FontColor black</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="136" x="15" y="359.4102">FontName Verdana</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="375.8984">}</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="392.3867">&#160;</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="292" x="5" y="408.875">skinparam backgroundColor transparent</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="425.3633">&#160;</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="99" x="5" y="441.8516">hide unlinked</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="458.3398">&#160;</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="474.8281">&#160;</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="491.3164">&#160;</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="507.8047">&#160;</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" text-decoration="wavy underline" textLength="189" x="5" y="524.293">actor Source as "" #orange</text><text fill="#FF0000" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="94" x="10" y="540.7813">Syntax Error?</text></g></svg></pre>
            <div onclick="location.href='#!';" id="4" class="overlay">
                <div class="popup" onclick="event.stopPropagation();">
                    <h2>PlantUml markup</h2>
                    <span class="copy" title="copy" onclick="copyToClipboard('sequence_uml_code4')"></span>
                    <a class="close" href="#!">&times;</a>
                    <div class="content">
                        <pre><code id="sequence_uml_code4">@startuml

!theme plain

skinparam backgroundColor transparent

hide unlinked




actor Source as &quot;&quot; #orange

participant Target as &quot;&quot; #powderblue



Source -&gt; Target: &lt;text fill=&quot;&quot;&gt;[[#1 {GET /path} GET /path]]&lt;/text&gt;

Target --&gt; Source: &lt;text fill=&quot;&quot;&gt;[[#2 {sync 200 response (25 ms)} sync 200 response (25 ms)]]&lt;/text&gt;


@enduml</code></pre>
                    </div>
                </div>
            </div>
            <sub><a href="#4">[source]</a></sub>
            <sub><a href="#5">[component diagram]</a></sub>
        </section>
        <!-- Popup for component diagram -->
        <div onclick="location.href='#!';" id="5" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>Component Diagram</h2>
                <a class="close" href="#!">&times;</a>
                <div class="content">
                    <pre><?xml version="1.0" encoding="us-ascii" standalone="no"?>
                        <svg xmlns="http://www.w3.org/2000/svg"
                             contentStyleType="text/css" height="550px" preserveAspectRatio="none"
                             style="width:304px;height:550px;background:#000000;" version="1.1" viewBox="0 0 304 550"
                             width="304px" zoomAndPan="magnify"><defs/><g><rect fill="#0E0C09" height="1" style="stroke:#0E0C09;stroke-width:1.0;" width="1" x="0" y="0"/><text fill="#33FF02" font-family="sans-serif" font-size="12" font-style="italic" font-weight="bold" lengthAdjust="spacing" textLength="116" x="5" y="17">PlantUML 1.2023.5</text><rect fill="#33FF02" height="21.4883" style="stroke:#33FF02;stroke-width:1.0;" width="292" x="5" y="27.1328"/><text fill="#000000" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="156" x="6" y="42.1328">[From string (line 6) ]</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="62.6211">&#160;</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="74" x="5" y="79.1094">@startuml</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="174" x="5" y="95.5977">!pragma layout smetana</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="112.0859">&#160;</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="128.5742">&#160;</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="145.0625">&#160;</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="9" x="5" y="161.5508">...</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="169" x="5" y="178.0391">... ( skipping 126 lines )</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="9" x="5" y="194.5273">...</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="211.0156">}</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="175" x="5" y="227.5039">skinparam StereotypeE {</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="168" x="15" y="243.9922">BackgroundColor white</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="129" x="15" y="260.4805">BorderColor black</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="276.9688">}</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="172" x="5" y="293.457">skinparam StereotypeI {</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="168" x="15" y="309.9453">BackgroundColor white</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="129" x="15" y="326.4336">BorderColor black</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="342.9219">}</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="178" x="5" y="359.4102">skinparam StereotypeN {</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="168" x="15" y="375.8984">BackgroundColor white</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="129" x="15" y="392.3867">BorderColor black</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="408.875">}</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="231" x="5" y="425.3633">skinparam UseCaseStereoType {</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="114" x="15" y="441.8516">FontColor black</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="136" x="15" y="458.3398">FontName Verdana</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="474.8281">}</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="292" x="5" y="491.3164">skinparam backgroundColor transparent</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="5" x="5" y="507.8047">&#160;</text><text fill="#33FF02" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" text-decoration="wavy underline" textLength="189" x="5" y="524.293">actor Source as "" #orange</text><text fill="#FF0000" font-family="sans-serif" font-size="14" font-weight="bold" lengthAdjust="spacing" textLength="94" x="10" y="540.7813">Syntax Error?</text></g></svg></pre>
                </div>
                <sub><a href="#uml5">[source]</a></sub>
            </div>
        </div>
        <!-- Popup for component diagram UML via component diagram popup -->
        <div onclick="location.href='#!';" id="uml5" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>PlantUml markup</h2>
                <span class="copy" title="copy" onclick="copyToClipboard('component_uml_code5')"></span>
                <a class="close" href="#5">&times;</a><!--show previous popup on close-->
                <div class="content">
                    <pre><code id="component_uml_code5">@startuml
!pragma layout smetana
!theme plain
skinparam backgroundColor transparent

actor Source as &quot;&quot; #orange
component Target as &quot;&quot; #powderblue


[Source] --&gt; Target


remove @unlinked
@enduml</code></pre>
                </div>
            </div>
        </div>
        <!-- Section containing data for sequence diagram popups -->
                <div onclick="location.href='#!';" id="1" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>GET /path</h2>
                <a class="close" href="#!">&times;</a>
                <div class="content">
                    <pre>{
  "requestHeaders": {
    "header": [ "value" ]
  },
  "body": "someBody"
}</pre>
                </div>
            </div>
        </div>
                <div onclick="location.href='#!';" id="2" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>sync 200 response (25 ms)</h2>
                <a class="close" href="#!">&times;</a>
                <div class="content">
                    <pre>{
  "responseHeaders": {
    "header": [ "value" ]
  },
  "body": "someBody"
}</pre>
                </div>
            </div>
        </div>
                </article>
</main>
</body>
<footer>
    <script type = "text/javascript" src = "https://d3js.org/d3.v4.min.js"></script>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.0.0/build/styles/default.min.css">
    <script src="https://lsd-consulting.github.io/lsd-core/src/main/resources/static/custom.js"></script>
    <script type = "text/javascript">
        addSliderForSvgZoom();
        highlightLifelinesWhenClicked();
    </script>
    <script src="https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.0.0/build/highlight.min.js"></script>
    <script>hljs.highlightAll();</script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js"></script>
    <script type = "text/javascript">
        highlightKeywords();                        highlightFact("3", "someTraceId");
                        unMarkSvg();
    </script>
</footer>
</html>
